design-2.md

```diff
@@ -0,0 +1,119 @@
+# Design-2
+
+A simple but bad program for the empty room would be as follows:
+
+    action go right
+    can move right {
+        move right
+        go right
+    }
+    action go up
+    can move up {
+        move up
+        go up
+    }
+
+    action go left
+    can move left {
+        move left
+        go left
+    }
+    action go right 2
+    can move right {
+        move right
+        go right 2
+    }
+    move down
+    go left
```

Comment:
Even after reading this several times I'm not entirely sure what it does. It seems like the line starting with "action" define functions, but these functions can be multiple words. I guess the next line starts the body of the function. The difference between "move" and "go" is confusing, especially because below you defined the action/function things as "goto"s. Or is it that every function must start with the word "go"? I'm also confused about the difference between the "go right 2" action, and how it is different from the "go right" action. It appears that the last two lines are basically how the program starts, but starting by moving down 1 square doesn't seem correct for any picobot program (what if you start at the bottom?) I'm also pretty confused about where the "else" case for the "can move ..." statements go.

In general, it seems like there's just too much syntax for programs that are as simple as picobot's. Maybe if you were trying to build a solver for some super intense map the controls and actions and stuff would all get used but for these simple examples it's just confusing me.

It's really cool that you're trying to make picobot an imperative language but I am just having trouble figuring out the syntax.